brian2tools/mdexport/expander.py

```diff
@@ -0,0 +1,842 @@
+                                                ) +
+                           self.check_plural(net_dict) +
+                           endl + horizontal_rule() + endl)
+        # start going to the dictionary items in particular run instance
+        for run_indx in range(len(net_dict)):
+            # details about the particular run
+            run_dict = net_dict[run_indx]
+            # start run header to say about duration
+            if len(net_dict) > 1:
+                run_string = (header('Run ' + str(run_indx + 1) +
+                              ' details', 3) +
+                              endl)
+            else:
+                run_string = endl
+            run_string += ('Duration of simulation is ' +
+                            bold(str(run_dict['duration'])) + endll)
+            # map expand functions for particular components
+            # h: normal 
+            func_map = {'neurongroup': {'f': self.expand_NeuronGroup,
+                                        'hb': 'NeuronGroup',
+                                        'h': 'Neuron group'},
+                       'poissongroup': {'f': self.expand_PoissonGroup,
+                                        'hb': 'PoissonGroup',
+                                        'h': 'Poisson spike source'},
+                       'spikegeneratorgroup':
+                                    {'f': self.expand_SpikeGeneratorGroup,
+                                     'hb': 'SpikeGeneratorGroup',
+                                     'h': 'Spike generating source'},
+                       'statemonitor': {'f': self.expand_StateMonitor,
+                                        'hb': 'StateMonitor',
+                                        'h': 'Activity recorder'},
+                       'spikemonitor': {'f': self.expand_SpikeMonitor,
+                                        'hb': 'SpikeMonitor',
+                                        'h': 'Spiking activity recorder'},
+                       'eventmonitor': {'f': self.expand_EventMonitor,
+                                        'hb': 'EventMonitor',
+                                        'h': 'Event activity recorder'},
+                       'populationratemonitor':
+                                    {'f': self.expand_PopulationRateMonitor,
+                                     'hb': 'PopulationRateMonitor',
+                                     'h': 'Population rate recorder'},
+                       'synapses': {'f': self.expand_Synapses,
+                                    'hb': 'Synapse',
+                                    'h': 'Synapse'},
+                       'poissoninput': {'f': self.expand_PoissonInput,
+                                         'hb': 'PoissonInput',
+                                         'h': 'Poisson input'}}
+            # loop through the components
+            for (obj_key, obj_list) in run_dict['components'].items():
```

Comment:
I think it would be clearer if we order the objects (it would also be good to reattach things like connects/initializers with their objects, but that's for another PR). E.g. first all neuron groups, then groups like `PoissonGroup` or `SpikeGeneratorGroup`, then `Synapses` and finally the monitor. Currently, the description might start with a monitor, then describe synapses, and only then the neurons that are referred by the earlier descriptions.

Reply:
>(it would also be good to reattach things like connects/initializers with their objects, but that's for another PR)

Yes sure. Also, some small things that are missing,